src/Oracle/Protocol/OracleHttpProtocol.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using JArray = Newtonsoft.Json.Linq.JArray;
+using JObject = Newtonsoft.Json.Linq.JObject;
+
+namespace Neo.Plugins
+{
+    internal class OracleHttpProtocol : IOracleProtocol
+    {
+        public const int Timeout = 5000;
+        public bool AllowPrivateHost { get; internal set; } = false;
+        public readonly string[] AllowedFormats = new string[] { "application/json" };
+
+        public byte[] Request(ulong requestId, string url, string filter)
+        {
+            Utility.Log(nameof(OracleHttpProtocol), LogLevel.Debug, $"Request: {url}");
+
+            Uri.TryCreate(url, UriKind.Absolute, out var uri);
+            using var handler = new HttpClientHandler
```

Comment:
We should avoid private servers

https://github.com/neo-project/neo/pull/1584/files#diff-0f0abffe8748b38e0c61e7c6a9436578R67-R72

Reply:
Done